app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt

```diff
@@ -154,6 +156,30 @@ class EventDetailsFragment : Fragment() {
             loadTicketFragment()
         }
 
+        eventViewModel.loadOrders()
+        eventViewModel.orders
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer {
+                it.forEach { order ->
+                    if (order.event?.id == safeArgs.eventId) {
+                        rootView.alreadyRegisteredLayout.visibility = View.VISIBLE
```

Comment:
You should also add a variable to retain the visibility state of alreadyRegisteredLayout in the ViewModel.
Just add var isAlreadyOrdered = false  in the viewModel-> change that to true when `order.event?.id == safeArgs.eventId` and set `rootView.alreadyRegisteredLayout.isVisible = viewModel.isAlreadyOrdered` inside onCreateView()

Reply:
> `rootView.alreadyRegisteredLayout.isVisible = viewModel.isAlreadyOrdered` inside onCreateView()

Okay, then how it will update when the value is set to true if I write it outside observer?

Reply:
If the null check is there for orders from the view model, then this another variable is not required.

Reply:
oh yeah that's true